data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/configuration/PrometheusSinkConfiguration.java

```diff
@@ -6,280 +6,135 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
 import jakarta.validation.constraints.NotNull;
-import org.apache.commons.lang3.StringUtils;
-import org.opensearch.dataprepper.model.configuration.PluginModel;
-import org.opensearch.dataprepper.plugins.sink.prometheus.util.PrometheusSinkUtil;
+import org.hibernate.validator.constraints.time.DurationMax;
+import org.hibernate.validator.constraints.time.DurationMin;
+import org.opensearch.dataprepper.aws.api.AwsConfig;
+import org.opensearch.dataprepper.plugins.codec.CompressionOption;
 
-import java.net.URL;
 import java.time.Duration;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
 
-public class PrometheusSinkConfiguration {
-
-    private static final int DEFAULT_UPLOAD_RETRIES = 5;
-
-    static final boolean DEFAULT_INSECURE = false;
-
-    private static final String S3_PREFIX = "s3://";
-
-    static final String SSL_KEY_CERT_FILE = "sslKeyCertChainFile";
-    static final String SSL_KEY_FILE = "sslKeyFile";
-    static final String SSL = "ssl";
-    static final String AWS_REGION = "awsRegion";
-
-    public static final String STS_REGION = "sts_region";
-
-    public static final String STS_ROLE_ARN = "sts_role_arn";
-    static final boolean DEFAULT_USE_ACM_CERT_FOR_SSL = false;
-    public static final int DEFAULT_ACM_CERT_ISSUE_TIME_OUT_MILLIS = 120000;
-    public static final String SSL_IS_ENABLED = "%s is enabled";
-
-    public static final Duration DEFAULT_HTTP_RETRY_INTERVAL = Duration.ofSeconds(30);
-
-    private static final String HTTPS = "https";
-
-    private static final String HTTP = "http";
-
-    private static final String AWS_HOST_AMAZONAWS_COM = "amazonaws.com";
+import static com.linecorp.armeria.common.MediaTypeNames.X_PROTOBUF;
 
-    private static final String AWS_HOST_API_AWS = "api.aws";
+public class PrometheusSinkConfiguration {
 
-    private static final String AWS_HOST_ON_AWS = "on.aws";
+    static final int DEFAULT_MAX_RETRIES = 5;
 
-    private static final String DEFAULT_ENCODING = "snappy";
+    private static final CompressionOption DEFAULT_ENCODING = CompressionOption.SNAPPY;
 
-    private static final String DEFAULT_CONTENT_TYPE = "application/x-protobuf";
+    private static final String DEFAULT_CONTENT_TYPE = X_PROTOBUF;
 
-    private static final String DEFAULT_REMOTE_WRITE_VERSION = "0.1.0";
+    static final String DEFAULT_REMOTE_WRITE_VERSION = "0.1.0";
+    private static final Duration DEFAULT_REQUEST_TIMEOUT = Duration.ofSeconds(60);
+    private static final Duration DEFAULT_CONNECTION_TIMEOUT = Duration.ofSeconds(60);
+    private static final Duration DEFAULT_IDLE_TIMEOUT = Duration.ofSeconds(60);
 
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsConfig awsConfig;
 
     @NotNull
     @JsonProperty("url")
     private String url;
```

Comment:
It looks like you removed the `insecure` flag. We should require that this `url` start with `https://` if you aren't going to keep that.

We should keep the `insecure_skip_verify` since this is helpful in many situations where there is not a root CA.

Reply:
Also, "root CA" is not applicable to AWS, right? Will support other auth mechanisms in a different PR